pkg/controller/controller.go

```diff
@@ -210,6 +225,55 @@ func createWorker(queue workqueue.RateLimitingInterface, resourceType string, ma
 	}()
 }
 
+func (c *controller) monitorConfigMap() {
+	// Cannot wait for the informer to push something into a queue.
+	// What we're waiting on may never exist without us configuring
+	// it, so we have to poll/ ask for it the first time to get it set.
+
+	// Can we ask 'through' an informer? Is it a writeback cache? I
+	// only ever want to monitor and be notified about one configmap
+	// in a hardcoded place.
+	glog.V(9).Info("cluster ID monitor loop enter")
+	cm, err := c.kubeClient.CoreV1().ConfigMaps("default").Get("cluster-info", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		m := make(map[string]string)
+		c.clusterIDLock.RLock()
+		m["id"] = c.clusterID
+		c.clusterIDLock.RUnlock()
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cluster-info",
+			},
+			Data: m,
+		}
+		// if we fail to set the id,
+		// it could be due to permissions
+		// or due to being already set while we were trying
+		if _, err := c.kubeClient.CoreV1().ConfigMaps("default").Create(cm); err != nil {
+			glog.Warningf("due to error %q, could not set clusterid configmap to %#v ", err, cm)
+		}
+	} else if err != nil {
+		// cluster id exists and is set
+		// get id out of cm
+		if id := cm.Data["id"]; "" != id {
+			c.clusterIDLock.Lock()
```

Comment:
I think it would be helpful to add a comment why this is using Lock vs RLock and why each is safe.